owl/symbol.scm

```diff
@@ -2,26 +2,41 @@
 ;;; Symbols
 ;;;
 
-
 (define-library (owl symbol)
 
-   (export symbol? symbol->string)
+   (export
+      string->symbol
+      symbol?
+      symbol=?
+      symbol->string
+      render-symbol)
 
    (import
       (owl defmac)
+      (only (owl list) all)
       (owl string)
-      (only (owl syscall) error))
+      (only (owl syscall) error interact))
 
    (begin
-      (define (symbol? x) (eq? (type x) type-symbol))
-
-      (define (symbol->string x) 
-         (if (eq? (type x) type-symbol)
-            (let ((str (ref x 1)))
-               (cond
-                  ((string=? str "")
-                     "||") ;; make empty symbols less invisible
-                  ((m/ / str) ;; fixme: doesn't quote internal |:s yet
-                     (string-append "|" str "|"))
-                  (else str)))
-            (error "Not a symbol: " x)))))
+
+      (define string->symbol
+         (H interact 'intern))
+
+      (define (symbol? x)
+         (eq? (type x) type-symbol))
+
+      (define (symbol=? x . lst)
+         (and (symbol? x) (all (C eq? x) lst)))
+
+      (define (symbol->string x)
+         (if (symbol? x)
+            (ref x 1)
+            (error "Not a symbol: " x)))
+
+      (define (render-symbol sym tl)
```

Comment:
|| syntax still requires a bit more work from both parsing and rendering side. Note for example, that (string->symbol "||") should not be '||, and |\x42;| should be *. This is a step in a good direction though.